pkg/cloudprovider/util/net.go

```diff
@@ -36,3 +37,19 @@ func CIDRToIPAndNetMask(ipv4 string) (string, string, int, error) {
 	netmask := fmt.Sprintf("%d.%d.%d.%d", ipNet.Mask[0], ipNet.Mask[1], ipNet.Mask[2], ipNet.Mask[3])
 	return ip.String(), netmask, size, nil
 }
+
+// GenerateRandMAC generates a random unicast and locally administered MAC address.
+func GenerateRandMAC() net.HardwareAddr {
+	buf := make([]byte, 6)
+	var mac net.HardwareAddr
+
+	_, err := rand.Read(buf)
+	if err != nil {
+	}
+
+	// Set locally administered addresses bit and reset multicast bit
```

Comment:
In the future we might want to preserve the vendor prefix, just to keep this in mind. For now it's ok.

Reply:
ok, leave as is for now, but keeping it in mind.